### ISSUE TITLE
Learnable Window

### ISSUE THREAD

**Issue by gunikavashisht13472** [KinWaiCheuk/nnAudio#108]
Could you please elaborate why you have not used Learnable_window in STFT , Mel Spectrograms and MFCC but used in their inverse counterparts?

**Comment by KinWaiCheuk** [comment]
The learnable kernels are, by default, disable in all STFT, Mel spectrograms, MFCC, and their inverse counterparts.

If you are referring to the argument `refresh_win` as shown below, it is not for learnable kernels. It is for recalculating the `window_sumsquare` for different audio lengths, which is essential to obtained a correct inverse. If all of your audio clips are of the same length, you can make `refresh_win=False` to speed up the calculation a little bit.
https://github.com/KinWaiCheuk/nnAudio/blob/e3ad18d4c345806aa42732ffc7b8f60b3ab0e071/Installation/nnAudio/Spectrogram.py#L47

**Comment by gunikavashisht13472** [comment]
Thank you for the answer. I want to use RNNS instead of CNNs in my maodel. Will this code work for RNNs too?

**Comment by KinWaiCheuk** [comment]
> Thank you for the answer. I want to use RNNS instead of CNNs in my maodel. Will this code work for RNNs too?

Yes it will work. nnAudio is just for spectrogram extraction using GPU. Once you have that spectrogram, you can use any model of your choice. The CNN in the [example](https://kinwaicheuk.github.io/nnAudio/intro.html#on-the-fly-audio-processing) is just a demonstration on how to use nnAudio together with a pytorch model.
